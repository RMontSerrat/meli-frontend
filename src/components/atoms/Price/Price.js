import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './Price.module.scss';
import { formatPrice, formatDecimals } from '~/utils';

const VARIANT_DEFAULT = 'default';
const VARIANT_BIG = 'big';
const variants = [VARIANT_DEFAULT, VARIANT_BIG];

const Price = ({
  currency, amount, decimals, variant,
}) => (
  <span className={classnames(styles.Price, styles[variants[variant]])}>
    <span className={styles.currency}>{currency}</span>
    <span className={styles.amount}>{`${formatPrice(amount)}`}</span>
    {decimals > 0 && (
    <span className={styles.decimals}>
      {formatDecimals(decimals)}
    </span>
    )}
  </span>
);

Price.propTypes = {
  currency: PropTypes.string.isRequired,
  amount: PropTypes.number.isRequired,
  decimals: PropTypes.number.isRequired,
  variant: PropTypes.oneOf(variants),
};

Price.defaultProps = {
  variant: VARIANT_DEFAULT,
};

export default Price;
