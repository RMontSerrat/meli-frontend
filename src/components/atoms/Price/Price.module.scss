.Price {
  display: inline-flex;
  font-family: $font-family;
  font-size: size(xbig);

  .currency {
    padding-right: rem(5px);
  }

  .decimals {
    font-size: size(small);
    line-height: 0.5rem;
    padding-left: rem(2px);
  }

  &.big {
    font-size: size(xxbig);

    .decimals {
      font-size: size(medium);
    }
  }
}
