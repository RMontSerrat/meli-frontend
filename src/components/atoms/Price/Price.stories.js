/* eslint-disable */
import React from 'react';
import { storiesOf } from '@storybook/react';
import Price from './Price';

storiesOf('Price', module).add('default', () => {
  const decimals = 8;
  const variantProp = 'default';

  return <Price currency="$" amount={10000} decimals={decimals} variant={variantProp} />
});
