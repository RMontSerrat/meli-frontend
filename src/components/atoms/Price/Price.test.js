import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Price from './Price';

describe('<Price />', () => {
  const priceProps = {
    currency: '$',
    amount: 10000,
    decimals: 11,
  };

  test('it should mount', () => {
    render(<Price {...priceProps} />);

    const price = screen.getByTestId('Price');

    expect(price).toBeInTheDocument();
  });

  test('snapshot with variatn default', () => {
    const { asFragment } = render(
      <Price {...priceProps} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('snapshot with variatn big', () => {
    const { asFragment } = render(
      <Price {...priceProps} variant="big" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
