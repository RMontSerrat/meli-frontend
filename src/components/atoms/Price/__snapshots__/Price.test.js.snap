// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Price /> snapshot with variatn big 1`] = `
<DocumentFragment>
  <span
    class="Price big"
    data-testid="Price"
  >
    <span
      class="currency"
    >
      $
    </span>
    <span
      class="amount"
    >
      10.000
    </span>
    <span
      class="decimals"
    >
      11
    </span>
  </span>
</DocumentFragment>
`;

exports[`<Price /> snapshot with variatn default 1`] = `
<DocumentFragment>
  <span
    class="Price default"
    data-testid="Price"
  >
    <span
      class="currency"
    >
      $
    </span>
    <span
      class="amount"
    >
      10.000
    </span>
    <span
      class="decimals"
    >
      11
    </span>
  </span>
</DocumentFragment>
`;
